### ISSUE TITLE
Add basic access control options for radicale

### ISSUE THREAD

**Issue by jvalleroy** [freedombox/Plinth#488]
Allow selection between "authenticated", "owner_only", or "owner_write" for rights type.

**Comment by harry-7** [comment]
I would like to work on this issue. Can you explain a bit more? as to what and where needs to updated?

**Comment by nikhilrayaprolu** [comment]
@harry-7 https://github.com/freedombox/Plinth/blob/master/actions/radicale#L59

**Comment by harry-7** [comment]
@nikhilrayaprolu  My doubt was should I add an option in the current main page or create a separate view? any comments on that ?

**Comment by jvalleroy** [comment]
@harry-7 I would suggest adding the option to the current page. Since it is a simple configuration, we can try to keep it to 1 page.

**Comment by harry-7** [comment]
The main page for Radicale seems a bit complicated I think I need to write a new main page :P . Is there any way to modify the original one to include this?

**Comment by jvalleroy** [comment]
@harry-7 I think one option is to extend plinth.forms.ServiceForm and plinth.views.ServiceView. So you would have a RadicaleServiceForm and RadicaleServiceView that inherit from those, but have the additional settings.

**Comment by harry-7** [comment]
@jvalleroy  So basically create a new form and use this form in the ServiceView and override a form_valid method? Is that what you mean ?

**Comment by jvalleroy** [comment]
@harry-7 I was thinking of something like this:
```
class RadicaleServiceForm(ServiceForm):
    """Specialized configuration form for radicale service."""
    access_control = forms.ChoiceField(...)

class RadicaleServiceView(ServiceView):
    """A specialized view for configuring radicale service."""
    form_class = forms.RadicaleServiceForm

    def get_initial(self):
        ...override method...

    def form_valid(self, form):
        ...override method...
```
But I haven't tested this, so feel free to make changes as needed.

**Comment by harry-7** [comment]
I was thinking of same thing  :) and I am trying it out but I don't get how to get the current rights type ?

**Comment by jvalleroy** [comment]
@harry-7 Similar to https://github.com/freedombox/Plinth/blob/master/actions/radicale#L59, you can use augeas to get the current value:
```
rights_type = aug.get('/files' + CONFIG_FILE + '/rights/type')
```

**Comment by jvalleroy** [comment]
Fixed in #532.
